.circleci/config.yml

```diff
@@ -0,0 +1,46 @@
+version: 2.1
+
+commands:
+  install-deps:
+    description: Install dependencies using pre-commit
+    steps:
+      - run:
+          command: |
+            pip install pre-commit tox flit
+            pre-commit install
+          name: Install dependencies
+
+jobs:
+  build:
+    docker:
+      - image: circleci/python:3.7
+    steps:
+      - checkout
+      - install-deps
+      - run: tox
+  deploy:
+    docker:
+      - image: circleci/python:3.7
+    steps:
+      - checkout
+      - install-deps
+      - run:
+          command: |
+            FLIT_USERNAME=$PYPI_USERNAME FLIT_PASSWORD=$PYPI_PASSWORD flit publish
+          name: Publish to Pypi
+
+workflows:
```

Comment:
Also add a PR workflow with a single job that runs pre-commit hooks, runs tox (tests), and builds with `flit`:

For the step that does the pre-commit hooks, here is talos's:
```yaml
- run:
    name: Run commit hooks
    command: |
      pre-commit run \
        --source "origin/${TARGET_BRANCH}" \
        --origin "origin/${CIRCLE_BRANCH}" \
        --show-diff-on-failure
```
CIRCLE_BRANCH is a default env var. You can set the TARGET_BRANCH env var by hitting the github api. See https://github.com/NarrativeScience/talos/blob/master/.circleci/commands/get-target-branch.yml#L7

Reply:
you can also use this orb! https://circleci.com/orbs/registry/orb/narrativescience/ghpr